pkg/operator/driverfeaturescontroller.go

```diff
@@ -0,0 +1,124 @@
+	manifest               []byte
+	kubeClient             kubernetes.Interface
+	operatorClient         v1helpers.OperatorClient
+	configMapLister        corelister.ConfigMapLister
+	infraLister            infralister.InfrastructureLister
+	clusterCSIDriverLister clustercsidriverlister.ClusterCSIDriverLister
+}
+
+func NewDriverFeaturesController(
+	name string,
+	namespace string,
+	manifest []byte,
+	kubeClient kubernetes.Interface,
+	kubeInformers v1helpers.KubeInformersForNamespaces,
+	operatorClient v1helpers.OperatorClient,
+	configInformer configinformers.SharedInformerFactory,
+	clusterCSIDriverInformer clustercsidriverinformer.ClusterCSIDriverInformer,
+	recorder events.Recorder,
+) factory.Controller {
+	configMapInformer := kubeInformers.InformersFor(namespace).Core().V1().ConfigMaps()
+	infraInformer := configInformer.Config().V1().Infrastructures()
+	c := &DriverFeaturesController{
+		name:                   name,
+		targetNamespace:        namespace,
+		manifest:               manifest,
+		kubeClient:             kubeClient,
+		configMapLister:        configMapInformer.Lister(),
+		operatorClient:         operatorClient,
+		infraLister:            infraInformer.Lister(),
+		clusterCSIDriverLister: clusterCSIDriverInformer.Lister(),
+	}
+	return factory.New().WithInformers(
+		configMapInformer.Informer(),
+		infraInformer.Informer(),
+		clusterCSIDriverInformer.Informer(),
+	).WithSync(
+		c.Sync,
+	).ResyncEvery(
+		time.Minute,
+	).WithSyncDegradedOnError(
+		operatorClient,
+	).ToController(
+		c.name,
+		recorder.WithComponentSuffix("feature-config-controlller-"+strings.ToLower(name)),
+	)
+}
+
+func (d DriverFeaturesController) Sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	opSpec, _, _, err := d.operatorClient.GetOperatorState()
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if opSpec.ManagementState != opv1.Managed {
+		return nil
+	}
+
+	// We will eventually need infra object to get the cluster defaults
+	// TODO: Read the defaults from infra object.
+
+	clusterCSIDriver, err := d.clusterCSIDriverLister.Get(utils.VSphereDriverName)
+	if err != nil {
+		return err
+	}
+
+	defaultFeatureConfigMap := resourceread.ReadConfigMapV1OrDie(d.manifest)
+
+	topologyCategories := utils.GetTopologyCategories(clusterCSIDriver)
+	if len(topologyCategories) > 0 {
+		requiredData := defaultFeatureConfigMap.Data
+		requiredData["improved-volume-topology"] = "true"
+		defaultFeatureConfigMap.Data = requiredData
+	}
+
+	_, _, err = resourceapply.ApplyConfigMap(ctx, d.kubeClient.CoreV1(), controllerContext.Recorder(), defaultFeatureConfigMap)
+	if err != nil {
+		return err
+	}
+	availableCondition := opv1.OperatorCondition{
+		Type:   d.name + opv1.OperatorStatusTypeAvailable,
+		Status: opv1.ConditionTrue,
+	}
+
+	progressingCondition := opv1.OperatorCondition{
+		Type:   d.name + opv1.OperatorStatusTypeProgressing,
+		Status: opv1.ConditionFalse,
+	}
```

Comment:
I think it's useless to set Progressing condition if it's never True. And IMO Available is quite useless too, this controlled could be fine just with Degraded, if anything goes wrong.
StaticResourceController does not have Available nor Progressing either.

Reply:
fixed